apps/base-docs/docs/building-with-base/guides/cross-chain-with-layerzero.md

```diff
@@ -0,0 +1,681 @@
+
+To install LayerZero smart contracts and their dependencies, run the following commands:
+
+```bash
+forge install GNSPS/solidity-bytes-utils --no-commit
+forge install OpenZeppelin/openzeppelin-contracts --no-commit
+forge install LayerZero-Labs/LayerZero-v2 --no-commit
+```
+
+Once installed, update your `foundry.toml` file by appending the following lines:
+
+```bash
+remappings = [
+    '@openzeppelin/contracts/=lib/openzeppelin-contracts/contracts',
+    'solidity-bytes-utils/=lib/solidity-bytes-utils',
+    '@layerzerolabs/lz-evm-oapp-v2/=lib/LayerZero-v2/oapp',
+    '@layerzerolabs/lz-evm-protocol-v2/=lib/LayerZero-v2/protocol',
+    '@layerzerolabs/lz-evm-messagelib-v2/=lib/LayerZero-v2/messagelib',
+]
+
+```
+
+---
+
+## Getting started with LayerZero
+
+LayerZero provides a smart contract standard called [OApp](https://docs.layerzero.network/contracts/oapp) that is intended for omnichain messaging and configuration.
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { OAppSender } from "./OAppSender.sol";
+import { OAppReceiver, Origin } from "./OAppReceiver.sol";
+import { OAppCore } from "./OAppCore.sol";
+
+abstract contract OApp is OAppSender, OAppReceiver {
+   constructor(address _endpoint) OAppCore(_endpoint, msg.sender) {}
+
+   function oAppVersion() public pure virtual returns (uint64 senderVersion, uint64 receiverVersion) {
+       senderVersion = SENDER_VERSION;
+       receiverVersion = RECEIVER_VERSION;
+   }
+}
+```
+
+:::info
+
+You can view the source code for this contract on [GitHub](https://github.com/LayerZero-Labs/LayerZero-v2/blob/main/oapp/contracts/oapp/OApp.sol).
+
+:::
+
+To get started using LayerZero, developers simply need to inherit from the [OApp](https://github.com/LayerZero-Labs/LayerZero-v2/blob/main/oapp/contracts/oapp/OApp.sol) contract, and implement the following two inherited functions:
+
+- `_lzSend`: A function used to send an omnichain message
+- `_lzReceive`: A function used to receive an omnichain message
+
+In this guide, you will be implementing the [OApp](https://docs.layerzero.network/contracts/oapp) standard into your own project to add the capability to send messages from a smart contract on Base to a smart contract on Optimism.
+
+:::info
+
+An extension of the [OApp](https://docs.layerzero.network/contracts/oapp) contract standard known as [OFT](https://docs.layerzero.network/contracts/oft) is also available for supporting omnichain fungible token transfers.
+
+:::
+
+:::info
+
+For more information on transferring tokens across chains using LayerZero, visit the [LayerZero documentation](https://docs.layerzero.network/contracts/oft).
+
+:::
+
+---
+
+## Writing the smart contract
+
+To get started, create a new Solidity smart contract file in your project's `src/` directory named `ExampleContract.sol`, and add the following content:
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
```

Comment:
Nit, but my understanding is that it's better to specify a specific version unless it's a library contract intended to be imported.